modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala

```diff
@@ -364,17 +364,17 @@ final class TcsEpics[F[_]: Sync](epicsService: CaService, tops: Map[String, Stri
 
   private val tcsStabilizeTime = 1.seconds
 
-  private val filteredInPositionAttr: CaAttribute[String] = new CaWindowStabilizer[String](inPositionAttr, java.time.Duration.ofMillis(tcsStabilizeTime.toMillis))
+  private val filteredInPositionAttr: CaWindowStabilizer[String] = new CaWindowStabilizer[String](inPositionAttr, java.time.Duration.ofMillis(tcsStabilizeTime.toMillis))
   def filteredInPosition:F[Option[String]] = safeAttribute(filteredInPositionAttr)
 
   // This functions returns a Task that, when run, will wait up to `timeout`
   // seconds for the TCS in-position flag to set to TRUE
-  def waitInPosition(timeout: Time): SeqAction[Unit] = waitForValue(filteredInPositionAttr, "TRUE", timeout,
+  def waitInPosition(timeout: Time): SeqAction[Unit] = waitForValue(filteredInPositionAttr.reset, "TRUE", timeout,
```

Comment:
I'm wondering about the `.reset` which mutates `filteredInPositionAttr` but happens outside of `IO`?  If you call `waitInPosition` while a previously created `SeqAction` instance is running for example what happens?

Reply:
You are right. It works now because the function happens to be called inside an `IO`, but it should be explicit.